### PR TITLE
drag/drop prevention, minimum width/heigths

### PR DESCRIPTION
**Pull request by Rdmkr**
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/345)

### CODE REVIEW

File: gui-js/apps/minsky-electron/src/app/managers/WindowManager.ts [hunk 1]
```diff
@@ -161,9 +161,20 @@ export class WindowManager {
     onCloseCallback?: (ev : Electron.Event) => void
   ) {
     const { width, height, title, modal = true, backgroundColor } = payload;
+
+    let minWidth = payload.minWidth;
+    if(!minWidth && width) minWidth = width;
+    if(!minWidth) minWidth = 300;
+
+    let minHeight = payload.minHeight;
+    if(!minHeight && width) minHeight = height;
```
Comment (highperformancecoder) [inline gui-js/apps/minsky-electron/src/app/managers/WindowManager.ts:170]: I think you mean height here, not width.

Comment (highperformancecoder) [reply gui-js/apps/minsky-electron/src/app/managers/WindowManager.ts:170]: Might this be cleaner to express as let minWidth=payload.minWidth? payload.minWidth: width? width: 300

**Review by highperformancecoder — CHANGES_REQUESTED** [review]

File: gui-js/apps/minsky-electron/src/app/managers/WindowManager.ts (hunk 1 above)
Comment (Rdmkr) [reply gui-js/apps/minsky-electron/src/app/managers/WindowManager.ts:170]: It's usually not recommended to use the ? : syntax twice in-line for readability... I settled on another notation using || though.

Comment (highperformancecoder) [reply gui-js/apps/minsky-electron/src/app/managers/WindowManager.ts]: That is brilliant. Sometimes you've got to love Javascript! Such an expression would be cast to bool in C++.

File: gui-js/libs/ui-components/src/lib/wiring/available-operations/available-operations.component.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, AfterViewInit } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (highperformancecoder) [reply gui-js/libs/ui-components/src/lib/wiring/available-operations/available-operations.component.ts:1]: We can fix this one later.

**Pull request merged**